### PR TITLE
Keras-like API website doc

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

@jason-dai Have a review?
The structure of new API website doc will be the following:

Keras-Style API Guide:
  - Python Guide   // General guide for Python users. Introduction + Sequential & Functional API + Methods + LeNet Example + Keras Code Support
  - Scala Guide   // General guide for Scala users. Introduction + Sequential & Functional API + Shape + Methods + LeNet Example
  - Layers:
    - Activations  // Available string representations for reference
    - Initializations  // Available string representations for reference
    - Core Layers // Dense (more to be added in later stage)
    - Other layer categories similar to Keras (more to be added in later stage)

### CODE REVIEW

File: docs/mkdocs.yml [hunk 1]
```diff
@@ -89,6 +89,9 @@ pages:
   - Full API docs:
     - Scala Docs: APIGuide/scaladoc.md
     - Python API Docs: APIGuide/python-api-doc.md
+- New API Guide:
+  - Layers:
+    - Core Layers: NewAPIGuide/Layers/core.md
```
Comment (jason-dai) [inline docs/mkdocs.yml]: Change to Keras-Style API Guide

File: docs/docs/NewAPIGuide/Layers/core.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+## Dense ##
```
Comment (jason-dai) [inline docs/docs/NewAPIGuide/Layers/core.md]: Make it clear that we now provide Keras 1.2 style APIs

File: docs/mkdocs.yml (hunk 1 above)
Comment (hkvision) [reply docs/mkdocs.yml]: Done.

File: docs/docs/NewAPIGuide/Layers/core.md (hunk 2 above)
Comment (hkvision) [reply docs/docs/NewAPIGuide/Layers/core.md]: Specified in https://github.com/intel-analytics/BigDL/pull/2342/files#diff-e55d3c2fc149d4f9b82b3db36e760e5eR3

**Comment by hkvision** [comment]
@jason-dai Have a review. The current structure will be clear enough for users to get familiar with our new API.

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+---
+## **Introduction**
+We hereby introduce a new set of __Keras-Style API__ based on [__Keras 1.2.2__](https://faroit.github.io/keras-docs/1.2.2/) in BigDL for the sake of user-friendliness. Users, especially those familiar with Keras, are recommended to use the new API to create a BigDL model and train, evaluate or tune it in a distributed fashion.
+
+To define a model in Python, now one just need to import the package
```
Comment (jason-dai) [inline docs/docs/KerasStyleAPIGuide/keras-api-python.md]: To define a model in Python using the Keras-style API

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+`bigdl.nn.keras.layer`.
```
Comment (jason-dai) [inline docs/docs/KerasStyleAPIGuide/keras-api-python.md]: remove the "." at the end

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+```
+
+
+---
+## **Layers**
+See [here](Layers/core.md) for all the available layers for the new set of Keras-Style API.
+
+
+---
+## **Methods**
```
Comment (jason-dai) [inline docs/docs/KerasStyleAPIGuide/keras-api-python.md]: It is not clear to me where these methods are defined: in layers?

File: docs/docs/KerasStyleAPIGuide/Layers/core.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
+---
```
Comment (jason-dai) [inline docs/docs/KerasStyleAPIGuide/Layers/core.md]: Delete this line?

File: docs/docs/KerasStyleAPIGuide/Layers/activation.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
+---
```
Comment (jason-dai) [inline docs/docs/KerasStyleAPIGuide/Layers/activation.md]: Delete this line?

File: docs/docs/KerasStyleAPIGuide/keras-api-scala.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
+---
```
Comment (jason-dai) [inline docs/docs/KerasStyleAPIGuide/keras-api-scala.md]: See the comments for the Python API doc

File: docs/docs/KerasStyleAPIGuide/Layers/core.md (hunk 6 above)
Comment (hkvision) [reply docs/docs/KerasStyleAPIGuide/Layers/core.md]: @jason-dai By adding this, there will be a segmentation line between two layers, which will make the layer look nicer.
See https://bigdl-project.github.io/master/#ScalaUserGuide/install-pre-built/ (with ---)
and https://bigdl-project.github.io/master/#ScalaUserGuide/install-build-src/ (without ---) for comparison. The former should look clearer.

**Comment by hkvision** [comment]
@jason-dai updated according to review comments both for Python doc and Scala doc~

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md (hunk 5 above)
Comment (hkvision) [reply docs/docs/KerasStyleAPIGuide/keras-api-python.md]: Added whether these methods are for layers or for models.

File: docs/docs/KerasStyleAPIGuide/Layers/activation.md (hunk 7 above)
Comment (hkvision) [reply docs/docs/KerasStyleAPIGuide/Layers/activation.md]: @jason-dai I fixed the style problem. It now shows correctly rendered by GitHub :)

**Comment by hkvision** [comment]
@jason-dai Updated according to the comments for style and refining the doc for methods.

**Comment by hkvision** [comment]
@jason-dai Merged. Thanks for the review!

**Pull request merged**